pkg/kubelet/kuberuntime/kuberuntime_container.go

```diff
@@ -611,13 +613,41 @@ func (m *kubeGenericRuntimeManager) killContainer(pod *v1.Pod, containerID kubec
 	return err
 }
 
+func isSidecar(containers []v1.Container, name string) bool {
+	for _, container := range containers {
+		if container.Name == name && lifecycleSidecar(container) {
+			return true
+		}
+	}
+	return false
+}
+
 // killContainersWithSyncResult kills all pod's containers with sync results.
 func (m *kubeGenericRuntimeManager) killContainersWithSyncResult(pod *v1.Pod, runningPod kubecontainer.Pod, gracePeriodOverride *int64) (syncResults []*kubecontainer.SyncResult) {
 	containerResults := make(chan *kubecontainer.SyncResult, len(runningPod.Containers))
 	wg := sync.WaitGroup{}
 
-	wg.Add(len(runningPod.Containers))
-	for _, container := range runningPod.Containers {
+	var containers []*kubecontainer.Container
```

Comment:
Seems like the decision for using pod graceful shutdown period while killing containers in 2 batches has been made such that both normal containers and sidecar containers shall have the same pod-level graceful shutdown periods. This will have the following effect:
1. pod can actually take **double** the graceful time to shutdown so semantically pod graceful shutdown time is not what it describes
2. users that heavily relies on this timeout will get affected as from their perspective, the behavior changed. (i.e. for node lifecycle management, user might have a script draining nodes with some timeout, this timeouts can also be used in hooks, etc at the cloud provider's level, it's a chain of things to affect)

have we decided that these effects are good with us? also, if so, I'd like to see that our api docs, and also release notes will need to reflect such change.

cc @tomwans @mattjmcnaughton @tedyu